Adding instructions for using the Airflow REST API locally. (#1073)
astronomer/docs: astro/airflow-api.md

You can use Airflow's [REST API](https://airflow.apache.org/docs/apache-airflow/stable/stable-rest-api-ref.html) to automate various Airflow workflows in your Deployments. For example, you can externally trigger a DAG run without accessing your Deployment directly by making an HTTP request in Python or cURL to the [corresponding endpoint](https://airflow.apache.org/docs/apache-airflow/stable/stable-rest-api-ref.html#operation/post_dag_run) in the Airflow REST API.

> 
> :::info

Reviewer: ```suggestion
```

Reviewer 2: Yea I'm cool with removing the admonition. If we keep it, I'd say to make it a `tip`

> To make an Airflow API request to the local instance of Airflow created with the Astro CLI, see this [section](https://docs.astronomer.io/astro/test-and-troubleshoot-locally#make-requests-to-the-airflow-rest-api) of our Test and troubleshoot locally docs.

Reviewer: ```suggestion
To test Airflow API calls in a local Airflow environment running through the Astro CLI, see [Test and Troubleshoot Locally](test-and-troubleshoot-locally.md#make-requests-to-the-airflow-rest-api).
```

Reviewer 2: Lowercase on the title?

> 
> :::

Reviewer: ```suggestion
```
